### PR TITLE
new command: create appcr

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/6847

This PR implements a new command to create app cr.

```yaml
$ architect create appcr --name test --app-name foo --app-version bar --catalog baz
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  creationTimestamp: null
  labels:
    app-operator.giantswarm.io/version: 1.0.0
  name: test
  namespace: giantswarm
spec:
  catalog: baz
  kubeConfig:
    inCluster: true
  name: foo
  namespace: giantswarm
  version: bar
status: {}
```

This PR replaces: https://github.com/giantswarm/opsctl/pull/412

### CODE REVIEW

File: cmd/create/appcr/flag.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package appcr
+
+func init() {
```
Comment (MarcelMue) [inline cmd/create/appcr/flag.go]: I know we do this here but I hate it with a passion.

File: cmd/create/command.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/giantswarm/architect/cmd/create/appcr"
+)
+
+var (
+	Cmd = &cobra.Command{
+		Use:   "create",
+		Short: "create all the things.",
```
Comment (MarcelMue) [inline cmd/create/command.go]: Something better here would be nice.

File: pkg/app/app.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+	return appCR
+}
+
+func Print(w io.Writer, format string, appCR *applicationv1alpha1.App) error {
+	var output []byte
+	var err error
+
+	switch format {
+	case "json":
```
Comment (kopiczko) [inline pkg/app/app.go]: Why do we need json output?

File: pkg/app/app.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/ghodss/yaml"
```
Comment (kopiczko) [inline pkg/app/app.go]: Didn't you want to use this magical k8s library because of empty fields?

File: pkg/app/error.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+import (
+	"github.com/giantswarm/microerror"
+)
+
+var wrongFormatError = &microerror.Error{
+	Kind: "wrongFormatError",
+}
+
+func IsWrongFormatError(err error) bool {
```
Comment (kopiczko) [inline pkg/app/error.go]: Not handled so it should be replaced with `executionFailedError`.

File: pkg/app/app.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Namespace is the namespace where App CRs are created.
+	namespace = "giantswarm"
+
+	labelAppOperatorVersion = "app-operator.giantswarm.io/version"
+	labelReleaseCyclePhase  = "release-operator.giantswarm.io/release-cycle-phase"
+	labelServiceType        = "giantswarm.io/service-type"
```
Comment (kopiczko) [inline pkg/app/app.go]: I don't see labelReleaseCyclePhase and labelServiceType being used.

File: pkg/app/app.go [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+	applicationv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// Namespace is the namespace where App CRs are created.
+	namespace = "giantswarm"
+
+	labelAppOperatorVersion = "app-operator.giantswarm.io/version"
```
Comment (kopiczko) [inline pkg/app/app.go]: Can we use string literal instead of the const? What is the added value here?

File: cmd/create/command.go (hunk 2 above)
Comment (TheoBrigitte) [reply cmd/create/command.go]: Going point, that was my first intention then I forgot

File: pkg/app/app.go (hunk 3 above)
Comment (TheoBrigitte) [reply pkg/app/app.go]: We do not but in case it's there.

File: pkg/app/app.go (hunk 4 above)
Comment (TheoBrigitte) [reply pkg/app/app.go]: Yes, it does the job correctly while the k8s `PrintObj` gave me a hard time with dependencies.

Comment (kopiczko) [reply pkg/app/app.go]: Nice 👍

File: pkg/app/app.go (hunk 6 above)
Comment (TheoBrigitte) [reply pkg/app/app.go]: removed those

File: pkg/app/app.go (hunk 7 above)
Comment (TheoBrigitte) [reply pkg/app/app.go]: Using string literal now, I originally copied the whole thing from release-operator,

File: pkg/app/error.go (hunk 5 above)
Comment (TheoBrigitte) [reply pkg/app/error.go]: done

File: cmd/create/appcr/flag.go (hunk 1 above)
Comment (TheoBrigitte) [reply cmd/create/appcr/flag.go]: I hate it too, so I changed it.

**Pull request merged**